Makefile

```diff
@@ -1,5 +1,20 @@
 -include Makefile.config
 
+current_opam_version := $(shell opam --version)
+expected_opam_version:= 2
```

Comment:
I think we need `opam` version >= 2.1 
https://github.com/mirage/mirage/blob/main/mirage.opam#L12